### DISCUSSION TITLE
Execute a specific business rule

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4350] [Questions] [answered]
Can we execute a specific business rule? We are asked to show a broken rule message when a user leaves some textboxes.
I know we have CheckRules(property) but that would execute all the rules for that property.
Kind Regards

**Comment by hurcane** [comment]
The rules engine has methods for all rules and rules per property, but not specific rules. What is the concern with running all rules for the property? Why is it important to only have the one rule execute?

**Reply by rockfordlhotka** [reply]
I agree. It should be possible to use short-circuiting and/or gateway rules to get any desired behavior.

**Reply by Chicagoan2016** [reply]
The property related to the textbox have several business rules like required, certain format etc. 
The user wants to alerted only about incorrect format (when they leave the text box).

**Reply by hurcane** [reply]
So they want no notification that the field is required? I would classify that request as a UI concern, and not a business rule concern. You still need the other rules to run to ensure a valid entity.

I would code an independent filter in the UI project to only show the requested broken rule names.

**Reply by Chicagoan2016** [reply]
@hurcane,  We do get some strange requests once in a while : ).
I could code that in UI.

**Reply by rockfordlhotka** [reply, marked as answer]
I really think this is a UI concern.

Your UI code can look at the broken rules collection and filter on the rule uri to show the rule(s) you need.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka I could iterate through 'BrokenRulesCollection' and find the business rule

**Comment by ajohnstone-ks** [comment]
@Chicagoan2016 Could you set the rule you want as the lowest priorty and then StopProcessing if it fails?
